**IMPORTING LIBRARIES**

In [1]:
import pandas as pd

**IMPORTING THE DATASET**

In [8]:
dataset = pd.read_csv("/breast_cancer.csv")
X = dataset.iloc[:,1:-1].values
y = dataset.iloc[:,-1].values   

**SPLITTING THE DATASET INTO THE TRAINING SET AND TEST SET**

In [9]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.2,random_state=0)

**TRAINING THE LOGISTIC REGRESSION MODEL ON THE TRAINING SET**

In [10]:
from sklearn.linear_model import LogisticRegression
classifier = LogisticRegression(random_state=0)   # creating a LR model
classifier.fit(X_train,y_train) # training of LR model by calling fit model from our classifier object




LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=100,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=0, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False)

**PREDICTING THE TEST SET RESULTS**

In [13]:
y_pred = classifier.predict(X_test)

**MAKING THE CONFUSION MATRIX**

In [17]:
# HOW MANY CORRECT PREDICTION ARE THERE AND HOW MANY INCORRECT PREDICTION IS THERE WE WILL MAKE IT
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test,y_pred)
print(cm)

[[84  3]
 [ 3 47]]


In [18]:
(84+47)/(84+47+3+3) # TO GET ACCURACY

0.9562043795620438

**COMPUTINNG THE ACCURACY WITH K-FOLD CROSS VALIDATION**

In [21]:
from sklearn.model_selection import cross_val_score
accuracy = cross_val_score(estimator= classifier,X = X_train , y= y_train, cv = 10)
print("Accuracy: {:.2f}%".format(accuracy.mean()*100))  #want only 2 decimal
print("Standard deviation: {:.2f}%".format(accuracy.std()*100))

Accuracy: 96.70%
Standard deviation: 1.97%


**SUMMARY:- AS WE CAN SEE HERE THAT OUR LOGISTIC REGRESSION MODEL IS 96.70% ACCURATE AND S.D IS VERY LESS SO WE CAN INFER THAT OUR MODEL IS PERFORMING WELL FOR BREAST CANCER PREDICTION AND THEREFORE IT CAN BE TOTALLY VALIDATED BY DOCTOR OR MEDICAL INSITITUTIONS  ** 